
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>laris.tools package &#8212; LARIS 0.9.0 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!--
    this give us a css class that will be invisible only if js is disabled
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="_static/custom.css?v=0330b353" />
  
  <!-- So that users can add custom icons -->
  <script src="_static/scripts/fontawesome.js?digest=8878045cc6db502f8baf"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf" />

    <script src="_static/documentation_options.js?v=39bb1c6d"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'laris.tools';</script>
    <script src="_static/sidebar_toggle.js?v=3b053ec6"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="laris.plotting package" href="laris.plotting.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search the docs ..."
         aria-label="Search the docs ..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class=" navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
  
    <p class="title logo__title">LARIS 0.9.0 documentation</p>
  
</a></div>
    
  </div>
  
  <div class=" navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="installation.html">
    Installation
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="tutorials.html">
    Tutorials
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="modules.html">
    API
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
        </div>
      
      
        <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/genecell/LARIS" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-square-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="installation.html">
    Installation
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="tutorials.html">
    Tutorials
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-internal" href="modules.html">
    API
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/genecell/LARIS" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-square-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
<h3><a href="index.html">Table of Contents</a></h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">API</a></li>
</ul>
</div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
      <div class="sidebar-primary-item">
<div id="ethical-ad-placement"
      class="flat"
      data-ea-publisher="readthedocs"
      data-ea-type="readthedocs-sidebar"
      data-ea-manual="true">
</div></div>
  </div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item">

<nav aria-label="Breadcrumb" class="d-print-none">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a href="index.html" class="nav-link" aria-label="Home">
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
    
    <li class="breadcrumb-item"><a href="modules.html" class="nav-link">API</a></li>
    
    
    <li class="breadcrumb-item"><a href="laris.html" class="nav-link">laris package</a></li>
    
    <li class="breadcrumb-item active" aria-current="page"><span class="ellipsis">laris.tools package</span></li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="module-laris.tools">
<span id="laris-tools-package"></span><h1>laris.tools package<a class="headerlink" href="#module-laris.tools" title="Link to this heading">#</a></h1>
<p>LARIS Tools Module (laris.tl)</p>
<p>Core analytical tools for ligand-receptor interaction analysis in spatial transcriptomics data.</p>
<p>This module contains public functions for:
- Preparing ligand-receptor integration scores with spatial diffusion
- Running the LARIS algorithm to identify spatially-specific LR interactions
- Computing cell type-specific interaction scores</p>
<p>Main Functions:
- prepareLRInteraction: Calculate LR interaction scores using spatial neighborhoods
- runLARIS: Identify spatially-specific LR pairs and compute cell type interactions</p>
<dl class="py function">
<dt class="sig sig-object py" id="laris.tools.prepareLRInteraction">
<span class="sig-prename descclassname"><span class="pre">laris.tools.</span></span><span class="sig-name descname"><span class="pre">prepareLRInteraction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">AnnData</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lr_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number_nearest_neighbors</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_rep_spatial</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'X_spatial'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">AnnData</span></span></span><a class="headerlink" href="#laris.tools.prepareLRInteraction" title="Link to this definition">#</a></dt>
<dd><p>Calculate ligand-receptor integration scores using spatial neighborhood information.</p>
<p>This function computes diffused ligand-receptor interaction scores by considering
the spatial context of each cell. It uses k-nearest neighbors to create a spatial
neighborhood graph and calculates element-wise multiplication of diffused ligand
and receptor expression levels.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>adata</strong> (<em>AnnData</em>) – Annotated data matrix containing gene expression and spatial coordinates.
Must have <cite>.obsm[use_rep_spatial]</cite> with spatial coordinates.</p></li>
<li><p><strong>lr_df</strong> (<em>pd.DataFrame</em>) – DataFrame containing ligand-receptor pairs with columns ‘ligand’ and ‘receptor’.</p></li>
<li><p><strong>number_nearest_neighbors</strong> (<em>int</em><em>, </em><em>default=10</em>) – Number of nearest neighbors to consider for spatial diffusion.</p></li>
<li><p><strong>use_rep_spatial</strong> (<em>str</em><em>, </em><em>default='X_spatial'</em>) – Key in adata.obsm containing spatial coordinates.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>New AnnData object containing ligand-receptor interaction scores.
- <cite>.X</cite>: Sparse matrix of LR interaction scores (cells × LR pairs)
- <cite>.var_names</cite>: Ligand-receptor pair names in format “ligand::receptor”
- <cite>.obs</cite>: Cell metadata copied from input adata
- <cite>.obsm</cite>: Spatial and other representations copied from input adata</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>AnnData</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">laris</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">la</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="c1"># Define ligand-receptor pairs</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lr_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
<span class="gp">... </span>    <span class="s1">&#39;ligand&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Tgfb1&#39;</span><span class="p">,</span> <span class="s1">&#39;Vegfa&#39;</span><span class="p">],</span>
<span class="gp">... </span>    <span class="s1">&#39;receptor&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Tgfbr1&#39;</span><span class="p">,</span> <span class="s1">&#39;Kdr&#39;</span><span class="p">]</span>
<span class="gp">... </span><span class="p">})</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="c1"># Calculate LR integration scores</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lr_adata</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">prepareLRInteraction</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">lr_df</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">lr_adata</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>  <span class="c1"># (n_cells, n_lr_pairs)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="laris.tools.runLARIS">
<span class="sig-prename descclassname"><span class="pre">laris.tools.</span></span><span class="sig-name descname"><span class="pre">runLARIS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">lr_adata</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">AnnData</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">adata</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">AnnData</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_rep</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'X_spatial'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_nearest_neighbors</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">27</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_repeats</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mu</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sigma</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_lowly_expressed</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">expressed_pct</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_cells_expressed_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_top_lr</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">4000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">by_celltype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">groupby</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CellTypes'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_rep_spatial</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'X_spatial'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number_nearest_neighbors</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mu_celltype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">expressed_pct_celltype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_lowly_expressed_celltype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mask_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1e-06</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">calculate_pvalues</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">layer_celltype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_neighbors_permutation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_permutations</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunk_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">50000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prefilter_fdr</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prefilter_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">score_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1e-06</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">spatial_weight</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_conditional_pvalue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">DataFrame</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">DataFrame</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#laris.tools.runLARIS" title="Link to this definition">#</a></dt>
<dd><p>Identify spatially-specific ligand-receptor interactions using LARIS algorithm.</p>
<p>LARIS (Ligand And Receptor Interaction in Spatial transcriptomics) identifies LR pairs
that show spatial specificity by comparing observed spatial correlation patterns
with randomized null distributions. When <cite>by_celltype=True</cite>, the function also
computes cell type-specific interaction scores with optional statistical testing.</p>
<p>This is the main analytical function of the LARIS package, providing:</p>
<ol class="arabic simple">
<li><p><strong>Spatial Specificity Analysis</strong>: Identifies LR pairs that show non-random
spatial co-localization patterns (higher scores = stronger spatial organization)</p></li>
<li><p><strong>Cell type-specific Scores</strong>: Integrates spatial specificity with cell type
expression specificity and spatial co-localization to identify which sender-
receiver cell type pairs are communicating via which LR pairs</p></li>
<li><p><strong>Statistical Testing</strong>: Optional permutation-based P values with FDR correction
to identify statistically significant interactions</p></li>
</ol>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>lr_adata</strong> (<em>AnnData</em>) – <p>AnnData object containing LR interaction scores from <cite>prepareLRInteraction()</cite>.</p>
<p>Required contents:
- <cite>.X</cite>: Diffused LR scores (cells × LR pairs)
- <cite>.var_names</cite>: LR pair names (“ligand::receptor”)
- <cite>.obsm[use_rep]</cite>: Spatial coordinates or other representation</p>
</p></li>
<li><p><strong>adata</strong> (<em>AnnData</em><em>, </em><em>optional</em>) – <p>Original annotated data matrix with gene expression and spatial information.
<strong>Required when `by_celltype=True`</strong>.</p>
<p>Required contents (when by_celltype=True):
- <cite>.obs[groupby]</cite>: Cell type annotations
- <cite>.obsm[use_rep_spatial]</cite>: Spatial coordinates
- <cite>.X</cite> or <cite>.layers[layer_celltype]</cite>: Gene expression</p>
</p></li>
<li><p><strong>use_rep</strong> (<em>str</em><em>, </em><em>default='X_spatial'</em>) – Key in <cite>lr_adata.obsm</cite> for coordinates to use in spatial specificity analysis.
Typically spatial coordinates, but could be other representations.</p></li>
<li><p><strong>n_nearest_neighbors</strong> (<em>int</em><em>, </em><em>default=10</em>) – Number of spatial neighbors for building the adjacency matrix in the
spatial specificity analysis. Larger values capture broader spatial patterns.</p></li>
<li><p><strong>random_seed</strong> (<em>int</em><em>, </em><em>default=27</em>) – Random seed for reproducibility of permutation tests.</p></li>
<li><p><strong>n_repeats</strong> (<em>int</em><em>, </em><em>default=3</em>) – Number of random permutations to generate the null distribution for
spatial specificity scoring. More repeats give more stable estimates.</p></li>
<li><p><strong>mu</strong> (<em>float</em><em>, </em><em>default=1</em>) – Regularization parameter for spatial specificity. Higher values penalize
interactions that look similar to random background more strongly.</p></li>
<li><p><strong>sigma</strong> (<em>float</em><em>, </em><em>default=100</em>) – Bandwidth parameter for Gaussian distance kernel in adjacency matrix.
Controls how quickly spatial weights decay with distance.</p></li>
<li><p><strong>remove_lowly_expressed</strong> (<em>bool</em><em>, </em><em>default=True</em>) – Whether to filter out LR pairs with low expression before ranking.</p></li>
<li><p><strong>expressed_pct</strong> (<em>float</em><em>, </em><em>default=0.1</em>) – Minimum fraction of cells expressing an LR pair (if remove_lowly_expressed=True).</p></li>
<li><p><strong>n_cells_expressed_threshold</strong> (<em>int</em><em>, </em><em>default=100</em>) – Minimum number of cells expressing an LR pair for it to be ranked.
Pairs below this threshold receive a penalty in ranking.</p></li>
<li><p><strong>n_top_lr</strong> (<em>int</em><em>, </em><em>default=4000</em>) – Number of top-ranked spatially-specific LR pairs to return.</p></li>
<li><p><strong>by_celltype</strong> (<em>bool</em><em>, </em><em>default=True</em>) – Whether to compute cell type-specific interaction scores. If False,
only returns spatial specificity results (much faster). If True,
<cite>adata</cite> must be provided.</p></li>
<li><p><strong>by_celltype=True</strong><strong>)</strong> (<em>Cell Type Analysis Parameters</em><em> (</em><em>only used if</em>)</p></li>
<li><p><strong>-------------------------------------------------------------</strong></p></li>
<li><p><strong>groupby</strong> (<em>str</em><em>, </em><em>default='CellTypes'</em>) – Column in <cite>adata.obs</cite> defining cell type groups.</p></li>
<li><p><strong>use_rep_spatial</strong> (<em>str</em><em>, </em><em>default='X_spatial'</em>) – Key in <cite>adata.obsm</cite> for spatial coordinates (for cell type analysis).</p></li>
<li><p><strong>number_nearest_neighbors</strong> (<em>int</em><em>, </em><em>default=10</em>) – Number of spatial neighbors for cell type co-localization analysis.</p></li>
<li><p><strong>mu_celltype</strong> (<em>float</em><em>, </em><em>default=100</em>) – Regularization parameter for COSG cell type specificity calculation.
Higher values more strongly penalize broadly expressed genes.</p></li>
<li><p><strong>expressed_pct_celltype</strong> (<em>float</em><em>, </em><em>default=0.1</em>) – Minimum expression fraction for cell type analysis.</p></li>
<li><p><strong>remove_lowly_expressed_celltype</strong> (<em>bool</em><em>, </em><em>default=True</em>) – Whether to filter lowly expressed genes in cell type analysis.</p></li>
<li><p><strong>mask_threshold</strong> (<em>float</em><em>, </em><em>default=1e-6</em>) – Numerical threshold for masking near-zero values.</p></li>
<li><p><strong>calculate_pvalues</strong> (<em>bool</em><em>, </em><em>default=True</em>) – Whether to perform permutation testing for statistical significance.
Set to False for faster exploratory analysis.</p></li>
<li><p><strong>layer_celltype</strong> (<em>str</em><em>, </em><em>optional</em>) – Layer in <cite>adata.layers</cite> to use for expression data. If None, uses <cite>adata.X</cite>.</p></li>
<li><p><strong>n_neighbors_permutation</strong> (<em>int</em><em>, </em><em>default=30</em>) – Number of similar interactions to use as background controls for
permutation testing. These are selected based on similarity of
diffused score profiles.</p></li>
<li><p><strong>n_permutations</strong> (<em>int</em><em>, </em><em>default=1000</em>) – Number of permutations for statistical testing. Common values:
- 1000: Quick testing
- 5000: More precise p-values
- 10000: Publication-quality precision</p></li>
<li><p><strong>chunk_size</strong> (<em>int</em><em>, </em><em>default=50000</em>) – Number of interactions to process simultaneously during permutation.
Larger values are faster but use more memory.</p></li>
<li><p><strong>prefilter_fdr</strong> (<em>bool</em><em>, </em><em>default=True</em>) – If True, only test interactions with scores &gt; prefilter_threshold for
significance. Others get FDR p-value = 1.0. This reduces multiple
testing burden and focuses power on high-scoring interactions.</p></li>
<li><p><strong>prefilter_threshold</strong> (<em>float</em><em>, </em><em>default=0.0</em>) – Minimum interaction score for FDR testing (if prefilter_fdr=True).</p></li>
<li><p><strong>score_threshold</strong> (<em>float</em><em>, </em><em>default=1e-6</em>) – Numerical precision threshold. Scores below this are set to exactly 0.0.</p></li>
<li><p><strong>spatial_weight</strong> (<em>float</em><em>, </em><em>default=1.0</em>) – Exponent applied to spatial specificity scores. Controls influence on
final interaction scores:
- 0: Ignore spatial specificity
- 1: Linear influence (default)
- &gt;1: Stronger emphasis on spatial specificity
- &lt;1: Weaker emphasis on spatial specificity</p></li>
<li><p><strong>use_conditional_pvalue</strong> (<em>bool</em><em>, </em><em>default=False</em>) – Use conditional p-value calculation for zero-inflated data. <strong>Recommended
for sparse datasets</strong>. When True:
- Interactions with score=0 get p-value=1.0
- Non-zero scores compared only to non-zero background
- Prevents spurious significance from sparse null distributions</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><dl>
<dt>If <cite>by_celltype=False</cite>:</dt><dd><p>Single DataFrame with spatial specificity results:</p>
<p>Columns:
- ‘ligand’: Ligand gene name
- ‘receptor’: Receptor gene name
- ‘score’: LARIS spatial specificity score
- ‘Rank’: Rank (0 = highest scoring)</p>
<p>Index: LR pair names (“ligand::receptor”)
Sorted by score (descending)</p>
</dd>
<dt>If <cite>by_celltype=True</cite>:</dt><dd><p>Tuple of (laris_lr, celltype_results) where:</p>
<ul>
<li><p><strong>laris_lr</strong>: DataFrame as described above</p></li>
<li><p><strong>celltype_results</strong>: DataFrame with cell type-specific scores:</p>
<p>Columns:
- ‘sender’: Cell type sending the ligand
- ‘receiver’: Cell type receiving the signal
- ‘ligand’: Ligand gene name
- ‘receptor’: Receptor gene name
- ‘interaction_name’: “ligand::receptor”
- ‘interaction_score’: Integrated LARIS score
- ‘p_value’: Raw permutation p-value (if calculate_pvalues=True)
- ‘p_value_fdr’: FDR-corrected p-value (if calculate_pvalues=True)
- ‘nlog10_p_value_fdr’: -log10(FDR) for visualization</p>
<p>Sorted by interaction_score (descending)</p>
</li>
</ul>
</dd>
</dl>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame or Tuple[pd.DataFrame, pd.DataFrame]</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>ValueError</strong> – If by_celltype=True but adata is not provided, or if required data
    is missing from adata or lr_adata.</p></li>
<li><p><strong>ImportError</strong> – If required helper functions are not available.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Examples</p>
<p><strong>Example 1: Quick spatial specificity analysis (no cell types)</strong></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">laris</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">la</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="c1"># Prepare LR scores</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lr_adata</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">prepareLRInteraction</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">lr_df</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="c1"># Identify spatially-specific LR pairs only</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">laris_lr</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">runLARIS</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">lr_adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">by_celltype</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">n_top_lr</span><span class="o">=</span><span class="mi">1000</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">laris_lr</span><span class="o">.</span><span class="n">head</span><span class="p">())</span>
</pre></div>
</div>
<p><strong>Example 2: Full analysis with cell type-specific scores</strong></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="c1"># Full LARIS analysis with cell types</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">laris_lr</span><span class="p">,</span> <span class="n">celltype_results</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">runLARIS</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">lr_adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">by_celltype</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">groupby</span><span class="o">=</span><span class="s1">&#39;cell_type&#39;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">calculate_pvalues</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">n_permutations</span><span class="o">=</span><span class="mi">5000</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="c1"># Filter for significant interactions</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sig_results</span> <span class="o">=</span> <span class="n">celltype_results</span><span class="p">[</span>
<span class="gp">... </span>    <span class="n">celltype_results</span><span class="p">[</span><span class="s1">&#39;p_value_fdr&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mf">0.05</span>
<span class="gp">... </span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Found </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">sig_results</span><span class="p">)</span><span class="si">}</span><span class="s2"> significant interactions&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Example 3: Fast exploratory analysis (no p-values)</strong></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">laris_lr</span><span class="p">,</span> <span class="n">celltype_results</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">runLARIS</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">lr_adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">by_celltype</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">calculate_pvalues</span><span class="o">=</span><span class="kc">False</span>  <span class="c1"># Much faster!</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Example 4: Conservative testing for sparse data</strong></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">laris_lr</span><span class="p">,</span> <span class="n">celltype_results</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">runLARIS</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">lr_adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">by_celltype</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">use_conditional_pvalue</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>  <span class="c1"># Robust for sparse data</span>
<span class="gp">... </span>    <span class="n">n_permutations</span><span class="o">=</span><span class="mi">5000</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">prefilter_fdr</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">prefilter_threshold</span><span class="o">=</span><span class="mf">0.01</span>  <span class="c1"># Only test score &gt; 0.01</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Example 5: Emphasize spatial specificity</strong></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">laris_lr</span><span class="p">,</span> <span class="n">celltype_results</span> <span class="o">=</span> <span class="n">la</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">runLARIS</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">lr_adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">adata</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">by_celltype</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">spatial_weight</span><span class="o">=</span><span class="mf">2.0</span>  <span class="c1"># Square the spatial scores</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><a class="reference internal" href="laris_1.tools.html#laris.tools.prepareLRInteraction" title="laris.tools.prepareLRInteraction"><code class="xref py py-obj docutils literal notranslate"><span class="pre">prepareLRInteraction</span></code></a></dt><dd><p>Prepare LR scores (prerequisite for this function)</p>
</dd>
</dl>
</div>
</dd></dl>

<p>..autofunction:: prepareLRInteraction
..autofunction:: runLARIS</p>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="laris.plotting.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">laris.plotting package</p>
      </div>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
<div
    id="pst-page-navigation-heading-2"
    class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
  </div>
  <nav class="bd-toc-nav page-toc" aria-labelledby="pst-page-navigation-heading-2">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#laris.tools.prepareLRInteraction"><code class="docutils literal notranslate"><span class="pre">prepareLRInteraction()</span></code></a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#laris.tools.runLARIS"><code class="docutils literal notranslate"><span class="pre">runLARIS()</span></code></a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf"></script>
<script defer src="_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
      © Copyright 2025, Min Dai, Tivadar Török, Dawei Sun, et al.
      <br/>
    
  </p>
</div>
      
        <div class="footer-item">

  <p class="sphinx-version">
    Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 7.3.7.
    <br/>
  </p>
</div>
      
    </div>
  
  
  
    <div class="footer-items__end">
      
        <div class="footer-item">
<p class="theme-version">
  <!-- # L10n: Setting the PST URL as an argument as this does not need to be localized -->
  Built with the <a href="https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html">PyData Sphinx Theme</a> 0.16.1.
</p></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>